rest/src/main/java/org/apache/unomi/rest/exception/InvalidRequestExceptionMapper.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.unomi.rest.exception;
+
+import org.apache.unomi.rest.validation.request.InvalidRequestException;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+
+@Provider
+@Component(service = ExceptionMapper.class)
+public class InvalidRequestExceptionMapper implements ExceptionMapper<InvalidRequestException> {
+    private static final Logger logger = LoggerFactory.getLogger(InvalidRequestExceptionMapper.class.getName());
+
+    @Override
+    public Response toResponse(InvalidRequestException exception) {
+        logger.error(exception.getMessage(), exception);
+
+        return Response.status(Response.Status.BAD_REQUEST).header("Content-Type", MediaType.TEXT_PLAIN)
+                .entity("Request rejected by the server because: " + exception.getResponseMessage()).build();
```

Comment:
I wonder if we should send the exception message because it might contain some sensitive data no?

Reply:
Maybe we could generate a debug log with more detailed information ?

Reply:
Forget what I said I was reading the code wrong about the message.